workshop_1/using_python_on_terminal.md

```diff
@@ -0,0 +1,208 @@
+
+![](images/storing_variable.png)
+
+You can overwrite variables so they will store something different. Let's try this by changing our greeting.
+
+![](images/overwrite_variable.png)
+
+We don't have to just store strings (strings are anything enclosed in quotation marks *"This is a string"*, *"?/+352%"* is also a string), we can store numbers too! This is handy because we can perform mathematical operations with them. Try storing some numbers as variables and performing some sums.
+
+![](images/numbers_as_variables.png)
+
+## Types
+
+String is a type of variable, and there a few variable types. Some languages require you to be explicit and always state what type of variable each variable is, but python will figure it out!
+
+![](images/variable_types.png)
+
+Types are important to know because we can't perform arithmetic operations on variables of different types. Try running this code:
+
+```
+greeting - a
+```
+
+![](images/unsupported_operand_types.png)
+
+Here: we get an error. Errors are really useful! We're told that this a "TypeError", which is do to with the variable types, and that it's because the operator "-" isn't supported for variables of type string (str) and integer (int).
+
+We can find out the type of a variable by running:
+
+```
+type(a)
+```
+
+## String manipulation
+
+While we can't adds strings and numbers togehter, we can add them to each other, and multiply them.
+
+```
+greeting = "Hi there, my name is"
+name = "Jennifer"
+```
+
+![](images/adding_strings.png)
+
+Note here how we had to add in an extra space to make the sentence make sense, when adding strings Python will just stick them together!
+
+
+We can also repeat strings - make sure you consider the spaces!
+
+```
+greeting = "'ello "
+greeting_end = "- what's all this then?"
+```
+
+![](images/repeating_strings.png)
+
+*Strings can be declared using single quotes too, but if you stick to double quotes you don't need to mess around in order to have apostrophes in your string*
+
+### F-strings
+
+What happens if we want to insert a number into our string? We can force an integer to change types to a string by using the "str()" method. This is called casting. Try our the example below.
+
+```
+balloons = 99
+print("I have " + str(balloons) + " red balloons!"
+```
+
+There's a better way to do this, using an f-string. F-strings (formatted strings) are a new-ish addition to python (it's always being improved!). It's a nice a way you can insert a variable into a string and this is much neater as you don't need to add multiple strings together.
+
+```
+balloons = 99
+print(f"I have {balloons} red balloons")
+```
+
+![](images/f_strings.png)
+
+
+
+### Applying methods to strings
+
+We can change the string too by applying a method (a method is something that is applied to an object (a variable is a type of object))
```

Comment:
a method is a function attached to an object (or rather the class the object is based on)